x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form.tsx

```diff
@@ -111,6 +113,44 @@ export function SloEditForm({ slo }: Props) {
   const { mutateAsync: createSlo, isLoading: isCreateSloLoading } = useCreateSlo();
   const { mutateAsync: updateSlo, isLoading: isUpdateSloLoading } = useUpdateSlo();
 
+  const createProcessedValues = () => {
+    const values = getValues();
+    if (isEditMode) {
+      return transformValuesToUpdateSLOInput(values);
+    }
+    return transformValuesToCreateSLOInput(values);
+  };
+
+  const handleCopyToJson = async () => {
+    const isValid = await trigger();
+    if (!isValid) {
+      return;
+    }
+    const values = createProcessedValues();
+    try {
+      await copyTextToClipboard(JSON.stringify(values, null, 2));
+      notifications.toasts.add({
+        title: i18n.translate('xpack.observability.slo.sloEdit.copyJsonNotification', {
+          defaultMessage: 'JSON copied to clipboard',
+        }),
+      });
+    } catch (e) {
+      notifications.toasts.add({
+        title: i18n.translate('xpack.observability.slo.sloEdit.copyJsonFailedNotification', {
+          defaultMessage: 'Could not copy JSON to clipboard',
+        }),
+      });
+    }
+  };
+
+  const copyTextToClipboard = async (text: string) => {
+    if (window.navigator?.clipboard) {
+      await window.navigator.clipboard.writeText(text);
+      return;
+    }
+    throw new Error('Could not copy to clipboard!');
```

Comment:
🍰 nit: inverse the condition to avoid the empty return.
```suggestion
    if (!window.navigator?.clipboard) {
       throw new Error('Could not copy to clipboard!');
    }
    await window.navigator.clipboard.writeText(text);
```